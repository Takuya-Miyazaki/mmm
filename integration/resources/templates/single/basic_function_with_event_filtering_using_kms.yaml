Resources:
  BasicFunctionWithEventFilteringUsingKmsKeyArn:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs18.x
      CodeUri: ${codeuri}
      MemorySize: 128
      Events:
        MySqsEvent:
          Type: SQS
          Properties:
            Queue: !GetAtt MySqsQueue.Arn
            FilterCriteria:
              Filters:
              - Pattern: '{ "body" : { "RequestCode" : [ "BBBB" ] } }'
            KmsKeyArn: !GetAtt MyKey.Arn

  MyKey:
    Type: AWS::KMS::Key
    Properties:
      Description: A sample key
      KeyPolicy:
        Version: '2012-10-17'
        Id: key-default-1
        Statement:
        - Sid: Allow administration of the key
          Effect: Allow
          Principal:
            AWS: !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:root
          Action:
          - kms:*
          Resource: '*'
        - Sid: Allow encryption/decryption access to Lambda Service Principal
          Effect: Allow
          Principal:
            Service: lambda.amazonaws.com
          Action: kms:Decrypt
          Resource: '*'

  MySqsQueue:
    Type: AWS::SQS::Queue

Metadata:
  SamTransformTest: true
