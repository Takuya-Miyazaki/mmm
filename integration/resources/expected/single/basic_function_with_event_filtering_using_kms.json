[
  {
    "LogicalResourceId": "BasicFunctionWithEventFilteringUsingKmsKeyArn",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "BasicFunctionWithEventFilteringUsingKmsKeyArnRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyKey",
    "ResourceType": "AWS::KMS::Key"
  },
  {
    "LogicalResourceId": "MySqsQueue",
    "ResourceType": "AWS::SQS::Queue"
  },
  {
    "LogicalResourceId": "BasicFunctionWithEventFilteringUsingKmsKeyArnMySqsEvent",
    "ResourceType": "AWS::Lambda::EventSourceMapping"
  }
]
